keps/sig-apps/3998-job-success-completion-policy/README.md

```diff
@@ -0,0 +1,860 @@
+	// The terminal state for such a job has the "Complete" condition.
+	// Additionally, these criteria are evaluated in order; Once the job meets one of the criteria,
+	// other criteria are ignored.
+	//
+	// +optional
+	Criteria []SuccessPolicyCriterion
+}
+
+// SuccessPolicyCriteria describes a criterion for declaring a Job as succeeded.
+// Each criterion must have at least one of the "succeededIndexes" or "succeededCount" specified.
+type SuccessPolicyCriterion struct {
+	// SuccessPolicyCriterion specifies the set of indexes 
+	// which need to be contained in the actual set of succeeded indexes for the job.
+	// The list of indexes must come within 0 to ".spec.completions" and
+	// must not contain duplicates. At least one element is required. 
+	// The indexes are represented as decimal integers
+	// separated by commas. The numbers are listed in increasing order. Three or
+	// more consecutive numbers are compressed and represented by the first and
+	// last element of the series, separated by a hyphen.
+	// For example, if the completed indexes are 1, 3, 4, 5 and 7, they are
+	// represented as "1,3-5,7".
+	// When this field is null, this field doesn't default to any value
+	// and is never evaluated at any time.
+	// 
+	// +optional
+	SucceededIndexes *string
+
+	// SucceededCount specifies the minimal size of the actual set of the succeeded indexes
+	// for the job.
+	// When this field is null, this field doesn't default to any value
+	// and is never evaluated at any time.
+	// When specified it needs to be a positive integer.
+	//
+	// +optional
+	SucceededCount *int
+	
+	// SetSuccessCriterionMetReason specifies the CamelCase reason for the "SuccessCriteriaMet" condition.
+	// Once the job meets this criterion, the specified reason is set to the "status.reason".
+	// The default value is "JobSuccessPolicy".
+	//
+	// +optional
+	SetSuccessCriterionMetReason *string
+}
+...
+
+// These are valid conditions of a job.
+const (
+	// JobSuccessCriteriaMet means the job has been succeeded.
+	JobSucceessCriteriaMet JobConditionType = "SuccessCriteriaMet"
+	...
+)
+```
+
+Moreover, we validate the following constraints for the `criteria`:
+- whether the specified indexes in the `succeededIndexes` and
+  the number of indexes in the `succeededCount` don't exceed the value of `completions`.
+- whether `Indexed` is specified in the `completionMode` field.
+- whether the size of `succeededIndexes` is under 64Ki.
+- whether the `succeededIndexes` field has a valid format.
+- whether the `succeededCount` field has an absolute number.
+- whether the criteria haven't changed.
+- whether the `setSuccessCriterionMetReason` filed has a valid CamelCase format.
+
+In addition, when the `successPolicy` isn't nil, we validate the `criteria` isn't nil.
+
+### Evaluation
+
+The job-controller evaluates the successPolicies follow the below rules in order every when pod conditions are updated:
```

Comment:
```suggestion
Every time the pod conditions are updated, the job-controller evaluates the successPolicies following the rules in order:
```
I think that's what you're trying to say here :sweat_smile:

Reply:
Yea, that's right. Your suggestions are much clearer :)

Reply:
Done.